test/flycheck-test.el

```diff
@@ -52,12 +52,10 @@
 (require 'flycheck-ert)
 
 ;; Make a best effort to make Coq Mode available
-;; (mapc (lambda (dir)
-;;         (add-to-list 'load-path (expand-file-name "coq/" dir)))
-;;       '("/usr/share/emacs/site-lisp/"
-;;         "/usr/local/share/emacs/site-lisp/"))
-(add-to-list 'load-path "/usr/share/emacs/site-lisp/")
-
+(mapc (lambda (dir)
+        (add-to-list 'load-path (expand-file-name "coq/" dir)))
+      '("/usr/share/emacs/site-lisp/"
+        "/usr/local/share/emacs/site-lisp/"))
```

Comment:
Good question; I just uncommented it, since it was here before.  I don't recall if it was strictly necessary for loading coq.  I'll check again.

Reply:
So, in this image (based on Ubuntu 18.04), `gallina.el` is indeed in `/usr/share/emacs/site-lisp/coq`.  This is not added because we run emacs with `-Q`, hence the code.

We could pass all quick options without `--no-site-lisp` and it should work locally.  It won't work on the image though, since Emacs is installed under `/opt/emacs` and it doesn't pick up the `/usr/share/emacs/site-lisp` there.

(There's a good reason to install Emacs under `/opt` instead of `/usr` in the Docker image: it's for grabbing the whole Emacs installation out of a Docker image and into another, without grabbing anything else we don't need)

So if we want to run the tests on CI in the current setup, it's a good idea to keep that code.  I will update the comment however.

Reply:
OK, makes sense. Alternatively, we could just `(setq major-mode coq-mode)` for the duration of the tests.